### PR TITLE
feat: create and attach configMap when adding file after commit

### PR DESCRIPTION
**Pull request by smuu**
Tested in branch: https://github.com/celestiaorg/knuu-example/tree/feature/add-file-after-commit

Resolves https://github.com/celestiaorg/knuu/issues/32

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by Bidon15 — CHANGES_REQUESTED** [review]
some nits - otherwise lgtm next round 👍

File: pkg/k8s/k8s_pod.go [hunk 1]
```diff
@@ -247,41 +262,83 @@ func buildEnv(envMap map[string]string) []v1.EnvVar {
...
+				},
 			},
-		},
+		}
+
+		podVolumes = append(podVolumes, podVolume)
 	}
 
-	return []v1.Volume{podVolume}, nil
+	defaultMode := int32(0777)
```
Comment (jrmanes) [inline pkg/k8s/k8s_pod.go:286]: `0777` 🥲 is it need it?, we're currently using [0550](https://github.com/celestiaorg/celestia-stack-internal/blob/main/data-availability/base/statefulset.yaml#L142) here

File: pkg/k8s/k8s_configmap.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	"context"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// GetConfigMap retrieves a configmap
+func GetConfigMap(namespace, name string) (*v1.ConfigMap, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_configmap.go]: https://github.com/celestiaorg/knuu/pull/154/commits/c5f8c59f74d8ba4b9ea4fad5fc0ad8d6e50c04a1

File: pkg/k8s/k8s_pod.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_pod.go:287]: https://github.com/celestiaorg/knuu/pull/154/commits/a777854923e02c4b6ccf50b867a1f4fc052b43e3

File: pkg/k8s/k8s_configmap.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	cm, err := Clientset().CoreV1().ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting configmap %s: %w", name, err)
+	}
+
+	return cm, nil
+}
+
+// ConfigMapExists checks if a configmap exists
+func ConfigMapExists(namespace, name string) (bool, error) {
```
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_configmap.go:27]: https://github.com/celestiaorg/knuu/issues/160

**Review by jrmanes — APPROVED** [review]
LGTM after talk 👌

**Pull request merged**